cumulus_library/study_parser.py

```diff
@@ -665,29 +668,62 @@ def _execute_build_queries(
     # Database exporting functions
 
     def export_study(
-        self, db: databases.DatabaseBackend, data_path: pathlib.Path
+        self,
+        db: databases.DatabaseBackend,
+        schema_name: str,
+        data_path: pathlib.Path,
+        archive: bool,
     ) -> list:
         """Exports csvs/parquet extracts of tables listed in export_list
-
         :param db: A database backend
-        :returns: list of executed queries (for unit testing only)
+        :param schema_name: the schema/database to target
+        :param data_path: the path to the place on disk to save data
+        :param archive: If true, get all study data and zip with timestamp
+        :returns: a list of queries, (only for unit tests)
         """
+        table_list = []
+        path = pathlib.Path(f"{data_path}/{self.get_study_prefix()}/")
         self.reset_counts_exports()
+        if archive:
+            table_query = base_templates.get_show_tables(
+                schema_name, f"{self.get_study_prefix()}__"
+            )
+            result = db.cursor().execute(table_query).fetchall()
+            for row in result:
+                table_list.append(row[0])
```

Comment:
nit: would feel more Pythonic to me to do `table_list = [row[0] for row in result]` and drop the `table_list = []` bit at the top. But I have a known bias for comprehensions :smile:

Reply:
something something jack of all trades, master of none/javascript poisoning something something - agree, will change